specification/sql/resource-manager/Microsoft.Sql/preview/2023-02-01-preview/JobPrivateEndpoints.json

```diff
@@ -0,0 +1,329 @@
+          },
+          "202": {
+            "description": "Accepted"
+          },
+          "204": {
+            "description": "The specified private endpoint does not exist."
+          }
+        },
+        "x-ms-long-running-operation": true,
+        "x-ms-long-running-operation-options": {
+          "final-state-via": "location"
+        },
+        "x-ms-examples": {
+          "Delete a private endpoint": {
+            "$ref": "./examples/DeleteJobPrivateEndpoint.json"
+          }
+        }
+      }
+    }
+  },
+  "definitions": {
+    "JobPrivateEndpoint": {
+      "description": "A job agent private endpoint.",
+      "type": "object",
+      "allOf": [
+        {
+          "$ref": "../../../common/v1/types.json#/definitions/ProxyResource"
+        }
+      ],
+      "properties": {
+        "properties": {
+          "$ref": "#/definitions/JobPrivateEndpointProperties",
+          "description": "Resource properties.",
+          "x-ms-client-flatten": true
+        }
+      }
+    },
+    "JobPrivateEndpointListResult": {
+      "description": "A list of job agent private endpoints.",
+      "type": "object",
+      "properties": {
+        "value": {
+          "description": "Array of results.",
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/JobPrivateEndpoint"
+          },
+          "readOnly": true
+        },
+        "nextLink": {
+          "description": "Link to retrieve next page of results.",
+          "type": "string",
+          "readOnly": true
+        }
+      }
+    },
+    "JobPrivateEndpointProperties": {
+      "description": "Properties of job agent private endpoint.",
+      "required": [
+        "targetServerAzureResourceId"
+      ],
+      "type": "object",
+      "properties": {
+        "targetServerAzureResourceId": {
+          "description": "ARM resource id of the server the private endpoint will target.",
+          "type": "string"
+        },
+        "privateEndpointId": {
```

Comment:
>privateEndpointId [](http://example.com/codeflow?start=9&length=17)

Is this also an ARM Resource? If yes, use this: [https://github.com/Azure/autorest/blob/main/docs/extensions/readme.md#x-ms-arm-id-details](https://github.com/Azure/autorest/blob/main/docs/extensions/readme.md#x-ms-arm-id-details).

Reply:
It is a system generated name of an ARM resource, but it is not the full ARM resource id. The example jsons might help clarify what I mean.